update SDK to v0.26.0

jpculp opened the pull request:
**Description of changes:**

* Rust: **1.58.1 → 1.61.0**
* Go: **1.16.13 → 1.18.2**
* govc: **0.27.1 → 0.28.0**
* openssl: **3.0.0 → 3.0.2-4**

**Testing done:**

- [x] Build SDK (and Bottlerocket) `x86_64` on `x86_64`
- [x] Build SDK (and Bottlerocket) `aarch64` on `x86_64`
- [x] Build SDK (and Bottlerocket) `aarch64` on `aarch64`
- [x] Build SDK (and Bottlerocket) `x86_64` on `aarch64`



**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

jpculp commented on Dockerfile:
```diff
@@ -473,16 +473,14 @@ ARG CGO_CXXFLAGS="${CXXFLAGS}"
 
 WORKDIR /home/builder/sdk-go/src
-RUN ./make.bash --no-clean
+RUN ./make.bash
```
`--no-clean` is deprecated and instead it is recommended to do `go install cmd`

jpculp commented on Dockerfile:
```diff
@@ -473,16 +473,14 @@ ARG CGO_CXXFLAGS="${CXXFLAGS}"
 # should use PIE, but any host binaries generated during the build
 # might not.
 WORKDIR /home/builder/sdk-go
-ENV GOOS="${GOOS}" \
-  GOROOT="/home/builder/sdk-go" \
-  GOPATH="/home/builder/gopath" \
```
This block was making things slightly more complex and ended up not really adding anything. Everything behaved normally when I removed it.

jpculp commented on Dockerfile:
```diff
@@ -714,8 +714,8 @@ COPY --chown=0:0 --from=sdk-cargo-deny /usr/libexec/tools/ /usr/libexec/tools/
 COPY --chown=0:0 --from=sdk-cargo-deny /usr/share/licenses/cargo-deny/ /usr/share/licenses/cargo-deny/
 
 # "sdk-govc" has the VMware govc tool and licenses.
-COPY --chown=0:0 --from=sdk-govc /usr/libexec/tools/govc /usr/libexec/tools/govc
-COPY --chown=0:0 --from=sdk-govc /usr/share/licenses/govmomi /usr/share/licenses/govmomi
```
This is just a consistency change.

jpculp committed:
update Rust to 1.61.0

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp committed:
address clippy warnings and update license-scan's Cargo.lock

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp committed:
update openssl to 3.0.2-4

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp commented:
Nearly everything had to be rebumped!

* Rust: **1.60.0 → 1.61.0**
* Go: **1.17.9 → 1.18.2**
* govc: **0.27.4 → 0.28.0**

Also added:
* Clippy fixes to `license-scan` and `license-tool`.
* Cherry-pick from https://github.com/bottlerocket-os/bottlerocket-sdk/pull/73 
_I'll rebase once that PR gets merged_

bcressey commented on hashes/openssl:
```diff
@@ -1,2 +1,2 @@
-# https://dl.fedoraproject.org/pub/fedora/linux/development/rawhide/Everything/source/tree/Packages/o/openssl-3.0.0-1.fc36.src.rpm
-SHA512 (openssl-3.0.0-1.fc36.src.rpm) = f0aa28ee6c2810e595310221a610c58861ef75b2c2002f6f65f715b25927fb3359b8dd3b0da7ce1d4578bfe9990f42190b8f4e94ed009ac53220dc381417ddf8
+# https://dl.fedoraproject.org/pub/fedora/linux/development/36/Everything/source/tree/Packages/o/openssl-3.0.2-4.fc36.src.rpm
+SHA512 (openssl-3.0.2-4.fc36.src.rpm) = 61d07f1d83e474d84bba10932d396d2099e5fe3424d0a7fc6af2617620b42e715c4d411093507868064156de7333b62c9425449ca83a12156fbe9992ea976b1b
```
I can't verify this URL - I only see "rawhide" on that Fedora server.

jpculp replied:
Whoops! I forgot to update the comment after Fedora 36 got released.

```suggestion
# https://dl.fedoraproject.org/pub/fedora/linux/releases/36/Everything/source/tree/Packages/o/openssl-3.0.2-4.fc36.src.rpm
SHA512 (openssl-3.0.2-4.fc36.src.rpm) = 61d07f1d83e474d84bba10932d396d2099e5fe3424d0a7fc6af2617620b42e715c4d411093507868064156de7333b62c9425449ca83a12156fbe9992ea976b1b
```

bcressey commented on Dockerfile:
```diff
@@ -492,8 +519,8 @@ RUN \
-  go install std && \
-  go install -buildmode=pie std
+  go install std cmd && \
+  go install -buildmode=pie std cmd
```
I'd guess that this is a slight behavior change in that we'll now end up with the `-buildmode=pie` versions of `go` and `gofmt` in the install directory. But I'm also not sure that matters, especially if the builds are fine.

bcressey reviewed (commented):
LGTM except for the OpenSSL package URL.

jpculp committed:
update govc to 0.28.0

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp committed:
update Go to 1.18.2

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp committed:
bump version to 0.26.0

Signed-off-by: Patrick J.P. Culp <jpculp@amazon.com>

jpculp commented:
- Fixed OpenSSL package URL.
_(SHA512 is still the same, it just changed directories)_
- Removed https://github.com/bottlerocket-os/bottlerocket-sdk/pull/73 in prep for rebase.

jpculp commented:
Rebased and ready for review!

cbgbt commented on license-scan/src/main.rs:
```diff
@@ -396,7 +396,7 @@ fn write_attribution(
             stated_license.to_string()
         } else {
-            licenses.sort();
+            licenses.sort_unstable();
```
This is fine, but I'm mostly just curious what prompted this change.

jpculp replied:
Nothing terrible. `clippy` just suggested the switch for performance reasons.

jpculp commented:
Rebased after https://github.com/bottlerocket-os/bottlerocket-sdk/pull/74 was merged.
